### PR TITLE
[receiver/skywalking] Add skywalking jvm metric receiver

### PR DESCRIPTION
**Pull request by aheling11**
**Description:** <Describe what has changed.>

Adding a JVM receiver in Skywalking and translate most of the metrics into the OpenTelemetry format according to the [OT JVM semantic conventions.](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/jvm/jvm-metrics.md) However, for the GC metrics, as the metric type is like gauge while the corresponding metric in the semantic conventions `jvm.gc.duration`is histogram type, I'm not translate it strictly according to the semantic conventions.
**Link to tracking Issue:** <Issue number if applicable>
#20315 
**Testing:** <Describe what testing was performed and which tests were added.>

test it locally. I export the metrics to Prometheus and display them in Grafana.
image

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/skywalkingreceiver/README.md [hunk 1]
```diff
@@ -40,5 +40,8 @@ service:
   pipelines:
     traces:
       receivers: [skywalking]
+    metrics:
+      receivers: [skywalking]
+
```
Comment (JaredTan95) [inline receiver/skywalkingreceiver/README.md:45]: In this PR, skywalking metrics only support `jvm` metrics, right? So, we need some notes to user. It does not support skywalking meter(https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Application-toolkit-meter.md).

**Review by JaredTan95 — CHANGES_REQUESTED** [review]

File: receiver/skywalkingreceiver/README.md (hunk 1 above)
Comment (aheling11) [reply receiver/skywalkingreceiver/README.md:45]: Yes, i will add some notes

File: receiver/skywalkingreceiver/factory.go [hunk 2]
```diff
@@ -37,7 +37,8 @@ func NewFactory() receiver.Factory {
 	return receiver.NewFactory(
 		metadata.Type,
 		createDefaultConfig,
-		receiver.WithTraces(createTracesReceiver, metadata.TracesStability))
+		receiver.WithTraces(createTracesReceiver, metadata.TracesStability),
+		receiver.WithMetrics(createMetricsReceiver, metadata.TracesStability))
```
Comment (jpkrohling) [inline receiver/skywalkingreceiver/factory.go:41]: It should be Metrics stability here.

Comment (aheling11) [reply receiver/skywalkingreceiver/factory.go]: done

**Comment by jiekun** [comment]
It seems the unit test and some checks are failed: 
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6167353114/job/16738196800
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6167353114/job/16738198910

Feel free to ask if you need a hand :) I'm glad to know that the metrics receiver is under review! Hope to get it merged soon and we would like to test it & have you some feedback.

**Comment by aheling11** [comment]
> It seems the unit test and some checks are failed:
> 
> * https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6167353114/job/16738196800
> * https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6167353114/job/16738198910
> 
> Feel free to ask if you need a hand :) I'm glad to know that the metrics receiver is under review! Hope to get it merged soon and we would like to test it & have you some feedback.

Thank you for your attention. I'll fix this.

**Review by JaredTan95 — APPROVED** [review]
Great job, LGTM

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**